src/mlpack/methods/adaboost/adaboost_main.cpp

```diff
@@ -202,8 +202,10 @@ int main(int argc, char *argv[])
     else if (weakLearner == "perceptron")
       m.WeakLearnerType() = AdaBoostModel::WeakLearnerTypes::PERCEPTRON;
 
+    const size_t numClasses = arma::max(labels) + 1;
```

Comment:
Ack, sorry, I thought this was in a different place than it is.  In this code the call to `NormalizeLabels()` forces the labels to be between 0 and numClasses.  But I can still make the call a little more efficient, because `numClasses == m.Mappings().n_elem`.